src/rust/sqs-executor/src/redis_cache.rs

```diff
@@ -136,53 +230,54 @@ impl RedisCache {
         if identities.is_empty() {
             return Ok(());
         }
-        let mut identities_to_check = Vec::with_capacity(identities.len());
-        {
+
+        let identities_to_check: Vec<_> = {
             let mut lru_cache = self.lru_cache.lock().unwrap();
-            for identity_bytes in identities.into_iter() {
-                if lru_cache.contains_key(&identity_bytes) {
-                    continue;
-                }
-                lru_cache.insert(identity_bytes.clone(), ());
-                identities_to_check.push(identity_bytes);
-            }
-        }
+
+            // for each of the identities:
+            //      1. update the lru cache status
+            //      2. if key is new (is_none()), collect into a Vec
+            identities
+                .into_iter()
+                .filter(|identity| lru_cache.put(identity.clone(), ()).is_none())
+                .collect()
+        };
+
         if identities_to_check.is_empty() {
             return Ok(());
         }
-        let identities = identities_to_check;
 
         let client_pool = self.connection_pool.clone();
         let task_start = Instant::now();
 
-        tokio::time::timeout(
-            Duration::from_millis(300),
-            tokio::spawn(async move {
-                let mut builder = MSetBuilder::new();
-                for identity in identities.iter() {
-                    builder = builder.set(identity, b"1");
-                }
-                let mut client = client_pool.get().await;
-                let res = client.mset(builder).await;
-                match res {
-                    Err(e) => {
-                        let elapsed = task_start.elapsed();
-                        if elapsed >= Duration::from_millis(300) {
-                            let elapsed_ms = elapsed.as_millis() as u64;
-                            error!(
-                                error = e.to_string().as_str(),
-                                elapsed_ms = elapsed_ms,
-                                "redis mset failed outside of ttl"
-                            );
-                            Ok(())
-                        } else {
-                            Err(e)
-                        }
+        tokio::spawn(async move {
+            let mut client = client_pool.get().await;
+            let mut mset_builder = MSetBuilder::new();
+
+            for identity in &identities_to_check {
+                mset_builder = mset_builder.set(identity, b"1");
```

Comment:
Is the significance of the `b"1"` that is used in a couple places just a flag to indicate presence (like how you make a HashMap behave as a Set)?

If so, it would be helpful to encode that as a constant (call it `SENTINEL_VALUE` or something similar, and document the intention with a docstring).

(If not, some documentation would still be useful 😅 )

Reply:
I also notice that a `()` appears to be used as a sentinel for the LRU cache, too, which is an interesting semantic duplication.

Which raises an interesting question: are we putting a cache in front of our cache?

Reply:
We have an LRU Cache (in memory) to determine if we should reach out to redis at all.